component/src/mock_client.rs

```diff
@@ -75,8 +78,39 @@ impl MockClient {
                         }
                     }
                 }
+                StatePayload::Swap { swap: payload, .. } => {
+                    match payload.trial_decrypt(&self.fvk) {
+                        Some(swap) => {
+                            self.nct.insert(Keep, payload.commitment)?;
+                            // At this point, we need to retain the swap plaintext,
+                            // and also derive the expected output notes so we can
+                            // notice them while scanning later blocks.
+                            self.swaps.insert(payload.commitment, swap.clone());
+
+                            let batch_data =
+                                block.swap_outputs.get(&swap.trading_pair).ok_or_else(|| {
+                                    anyhow::anyhow!("server gave invalid compact block")
+                                })?;
+
+                            let (output_1, output_2) = swap.output_notes(batch_data);
+                            // Pre-insert the output notes into our notes table, so that
+                            // we can notice them when we scan the block where they are claimed.
+                            self.notes.insert(output_1.commit(), output_1);
+                            self.notes.insert(output_2.commit(), output_2);
+                        }
+                        None => {
+                            self.nct.insert(Forget, payload.commitment)?;
+                        }
+                    }
+                }
                 StatePayload::RolledUp(commitment) => {
-                    self.nct.insert(Forget, commitment)?;
+                    if self.notes.contains_key(&commitment) {
+                        // This is a note we anticipated, so retain its auth path.
+                        self.nct.insert(Keep, commitment)?;
+                    } else {
+                        // This is someone else's note.
+                        self.nct.insert(Forget, commitment)?;
+                    }
```

Comment:
... so that when the `SwapClaim` generates a `RolledUp(commitment)`, we can recognize it as the output notes we already precomputed.

Reply:
(It's a moot point for the mock client, but we don't want the sync process to automatically generate a claim, though, because there's no way for the sync process to know whether or not a claim has already been submitted until reaching the chain tip, and submitting a duplicate one will leak information.)